src/sonic_ax_impl/mibs/vendor/cisco/ciscoSwitchQosMIB.py

```diff
@@ -132,8 +136,10 @@ def update_stats(self):
             namespace = self.port_index_namespace[if_index]
 
             # The first half of queue id is for ucast, and second half is for mcast
-            # To simulate vendor OID, we wrap queues by half distance
-            pq_count = math.ceil((max(if_queues) + 1) / 2)
+            # To simulate vendor OID, we wrap queues by max priority groups
+            port_max_queues = self.statedb.get_all(self.statedb.STATE_DB,
+                                                   mibs.buffer_max_parm_table(self.oid_name_map[if_index]))['max_queues']
+            pq_count = math.ceil(int(port_max_queues) / 2)
```

Comment:
Why is this halving done when now? There could be a possibility of MC queue not present, right?

Reply:
Yes, there is a possibility of MC queue not present.
We don't want to change the behavior, the purpose of  the halving here is to send the MC counter no on 9-16 rather on 1-8.
If there are MCs - they will be added to the same OID as before, and if there are no MCs, the following for loop will not reach a queue higher then pq_count

Reply:
what will be the value of port_max_queues if there are Unicast and multicast queues ?